common/crypto/platform/openssl/crypto.cpp

```diff
@@ -93,6 +93,45 @@ string GetOpenSSLErrorMessage() {
 	return errorDescription;
 }
 
+// We need our own custom callback routine, as the default one will prompt for a
+// passphrase.
+int PrivateKey::CapturePrivateKeyPassphrase(char *buf, int size, int rwflag, void *u) {
+	// We'll only use this callback for reading passphrases, not for writing
+	// them.
+	assert(rwflag == 0); 
+
+	// Copy passphrase to out, out is not expected to be null terminated.
+	char *pass = (char *)u;
+	strncpy(buf, pass, size);
+	return (static_cast<int>(strlen(pass)) < size) ? static_cast<int>(strlen(pass)) : size;
+}
+
+ExpectedPrivateKey PrivateKey::LoadFromPEM(const string &private_key_path, const string &passphrase) {
+	auto private_bio_key = unique_ptr<BIO, void (*)(BIO *)>(
+		BIO_new_file(private_key_path.c_str(), "r"), bio_free_func);
+	if (!private_bio_key.get()) {
+		return expected::unexpected(MakeError(
+			SetupError, "Failed to open the private key file: " + GetOpenSSLErrorMessage()));
+	}
+
+	// Maybe there is a better way to convert a string reference into a mutable
+	// C-string?
```

Comment:
Yes :)

`Cpp` strings have the `c_str()` function :)

Reply:
Yes, but it's not mutable. I'd use `vector<char>`, and it's `data()` function, which is mutable.

Reply:
You should also probably use [`copy`](https://en.cppreference.com/w/cpp/algorithm/copy) then, since again, it's safer than its C-counterpart.

Reply:
True, didn't think about the mutable part